src/Chapter2.hs

```diff
@@ -349,8 +374,11 @@ Implement a function that returns only the first half of a given list.
 "b"
 -}
 -- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf l = 
+   subList 0 (mid - 1) l
```

Comment:
Nice work on reusing `subList`! Though, sublist does more work than required here, and it would be enough to just _take_ first half of the list :slightly_smiling_face: